sos/plugins/sclnpm.py

```diff
@@ -0,0 +1,15 @@
+from sos.plugins import Plugin, SCLPlugin
+
+
+class SCLNpm(Plugin, SCLPlugin):
+    """
+    Get info about global modules from SCL collections that contain "npm"
+    """
+
+    requires_root = False
+    packages = ("%(scl_name)s-npm",)
+    profiles = ('system',)
+
+    def setup(self):
+        for scl in self.scls_matched:
+            self.add_cmd_output_scl(scl, "npm ls -g --json")
```

Comment:
Is this expected to be the "normal" way an `SCLPlugin` requests command output?

If that's the case then perhaps it would be better to push the `for scl in self.scls_matched` down into `Plugin.add_cmd_output_scl()`? That way all plugins that want the "loop over matched SCLs" behaviour can just call the method without any need for a loop. A new optional kwarg (e.g. `scl=None`) could then be added so that any plugins that *do* want control over this can still exercise it.